base/workflow/R/run.write.configs.R

```diff
@@ -93,13 +97,22 @@ run.write.configs <- function(settings, write = TRUE, ens.sample.method = "unifo
   scipen <- getOption("scipen")
   options(scipen = 12)
   
-  PEcAn.uncertainty::get.parameter.samples(settings, posterior.files, ens.sample.method)
   samples.file <- file.path(settings$outdir, "samples.Rdata")
   if (file.exists(samples.file)) {
     samples <- new.env()
     load(samples.file, envir = samples) ## loads ensemble.samples, trait.samples, sa.samples, runs.samples, env.samples
     trait.samples <- samples$trait.samples
-    ensemble.samples <- samples$ensemble.samples
+    trait_sample_indices <- input_design[["param"]]
+    ensemble.samples <- list()
+    for (pft in names(trait.samples)) {
+        pft_traits <- trait.samples[[pft]]
+        ensemble.samples[[pft]] <- as.data.frame(
+        lapply(names(pft_traits), function(trait) {
+          pft_traits[[trait]][trait_sample_indices]
+        })
+  )
+  names(ensemble.samples[[pft]]) <- names(pft_traits)
+}
```

Comment:
I had a hard time following the nesting here -- please keep closing brackets and parens indented the same amount as the line that opened them

```suggestion
    for (pft in names(trait.samples)) {
        pft_traits <- trait.samples[[pft]]
        ensemble.samples[[pft]] <- as.data.frame(
          lapply(
            names(pft_traits),
            function(trait) pft_traits[[trait]][trait_sample_indices]
          )
        )
      names(ensemble.samples[[pft]]) <- names(pft_traits)
    }
```

Reply:
Also: What you wrote here works, but if I'm following right you could also simplify this block to

```
    for (pft in names(trait.samples)) {
        pft_traits <- trait.samples[[pft]]
        ensemble.samples[[pft]] <- purrr::map_dfc(
            pft_traits,
            \(trait) trait[trait_sample_indices]
        )
    }
```